### PR TITLE
feat: [HL-92] Login <> Handle and integrate login use case 🍕

### PR DESCRIPTION
**Pull request by esraakhaled**
add mock Data for LoginUseCase until Actual Request is finished

#### Proposed Changes

- Add LoginUseCase but use Mock LoginRequest 
-  Add entity For returning User
-  Handle error if sign-in fail
- Show Loading and Stop Loading
- Add unit test for LoginUseCase

#### Related Issues

https://github.com/motoon-eg/healthy/pull/97

#### Additional Information

https://motoon-eg.atlassian.net/jira/software/projects/HL/boards/2?selectedIssue=HL-92
#### Screenshot

N/A

#### Checklist

- [x] I have tested these changes locally.
- [ ] I have added appropriate documentation or updated existing documentation.
- [x] I have added appropriate test coverage or updated existing test coverage.
- [ ] I have updated the changelog (if applicable).
- [x] I have followed the project's code style and formatting guidelines.
- [x] I have reviewed and adhered to the project's contributing guidelines.

### CODE REVIEW

File: Domain/.gitignore
Comment (ahmdmhasn) [inline Domain/.gitignore:1]: To be removed. Let's depend on the root `.gitignore`

File: Domain/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Domain
+
+A description of this package.
```
Comment (ahmdmhasn) [inline Domain/README.md]:
```suggestion
The Domain Layer is a crucial component of a software architecture that represents the core business logic and rules of the application. It acts as the heart of the system, encapsulating all the domain-specific logic and entities.
```

File: Domain/Sources/Domain/UseCases/LoginUseCase.swift [hunk 2]
```diff
@@ -0,0 +1,4 @@
+@available(iOS 13.0.0, *)
```
Comment (ahmdmhasn) [inline Domain/Sources/Domain/UseCases/LoginUseCase.swift]: Let's make the package support iOS 13 by default.

File: Healthy/AppCoordinator.swift
Comment (ahmdmhasn) [inline Healthy/AppCoordinator.swift:1]: Why did you move the AppCoordinator & AppDelegate outside classes?

File: Healthy/Classes/Modules/Onboarding/Login/LoginViewController.swift [hunk 3]
```diff
@@ -35,7 +35,6 @@ final class LoginViewController: UIViewController {
 
     override func viewDidLoad() {
         super.viewDidLoad()
-
         configureAppearance()
```
Comment (ahmdmhasn) [inline Healthy/Classes/Modules/Onboarding/Login/LoginViewController.swift]:
```suggestion

        configureAppearance()
```
I prefer keeping a line between view did load and configuration logic.

File: Healthy/Classes/Modules/Onboarding/Login/LoginViewController.swift [hunk 4]
```diff
@@ -116,14 +115,20 @@ private extension LoginViewController {
...
-            .sink { _ in
-                // TODO: Show error message.
+        viewModel.errorPublisher
+            .sink { error in
+                let alertController = UIAlertController(
+                    title: "Error!!",
+                    message: error.localizedDescription,
+                    preferredStyle: .alert)
+
+                self.present(alertController, animated: true)
```
Comment (ahmdmhasn) [inline Healthy/Classes/Modules/Onboarding/Login/LoginViewController.swift]:
```suggestion
            .sink { [weak self] error in
                let alertController = UIAlertController(
                    title: "Error!!",
                    message: error.localizedDescription,
                    preferredStyle: .alert)

                self?.present(alertController, animated: true)
```
Retain cycle.

**Review by ahmdmhasn — APPROVED** [review]

**Review by abdallah-7698 — APPROVED** [review]

File: Healthy/AppCoordinator.swift
Comment (esraakhaled) [reply Healthy/AppCoordinator.swift:1]: It was By Mistake trying to sort them by name

File: Healthy.xcodeproj/project.pbxproj
Comment (esraakhaled) [inline Healthy.xcodeproj/project.pbxproj:1]: solve issue  The following build commands failed: PhaseScriptExecution NewRelic /Users/runner/Library/Developer/Xcode/DerivedData/Healthy-ekcbpzeeypxcrkemmowbsefugrvm/Build/Intermediates.noindex/Healthy.build/Debug-iphonesimulator/Healthy.build/Script-025511B82A3C624E00295B91.sh ios swift (in target 'Healthy' from project 'Healthy') (1 failure) Error: Process completed with exit code 65. in CI/CD
